File: flutter_app/lib/models/quiz.dart
class Quiz {
  final String
  id;
  final String
  title;
  final String
  subject;
  final int
  year;
  final List<
    Question
  >
  questions;
  final String
  difficulty;

  Quiz({
    required this.id,
    required this.title,
    required this.subject,
    required this.year,
    required this.questions,
    required this.difficulty,
  });

  factory Quiz.fromJson(
    Map<
      String,
      dynamic
    >
    json,
  ) {
    return Quiz(
      id: json['_id'],
      title: json['title'],
      subject: json['subject'],
      year: json['year'],
      questions:
          (json['questions']
                  as List)
              .map(
                (
                  q,
                ) => Question.fromJson(
                  q,
                ),
              )
              .toList(),
      difficulty: json['difficulty'],
    );
  }
}

class Question {
  final String
  question;
  final List<
    String
  >
  options;
  final int?
  correctAnswer;
  final String?
  explanation;

  Question({
    required this.question,
    required this.options,
    this.correctAnswer,
    this.explanation,
  });

  factory Question.fromJson(
    Map<
      String,
      dynamic
    >
    json,
  ) {
    return Question(
      question: json['question'],
      options:
          List<
            String
          >.from(
            json['options'],
          ),
      correctAnswer: json['correctAnswer'],
      explanation: json['explanation'],
    );
  }
}

File: backend/src/models/Quiz.js
const mongoose = require('mongoose');

const quizSchema = new mongoose.Schema({
  title: {
    type: String,
    required: true,
  },
  subject: {
    type: String,
    required: true,
  },
  year: {
    type: Number,
    required: true,
  },
  questions: [{
    question: String,
    options: [String],
    correctAnswer: Number, // Index of correct option
    explanation: String,
  }],
  difficulty: {
    type: String,
    enum: ['easy', 'medium', 'hard'],
    default: 'medium',
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
});

module.exports = mongoose.model('Quiz', quizSchema);
